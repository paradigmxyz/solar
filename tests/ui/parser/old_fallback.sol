// https://github.com/foundry-rs/foundry/issues/9349

pragma solidity >=0.4.22 <0.6;

contract BugReport {
    function() external payable { //~ ERROR: expected a state variable declaration
        deposit();
        uint
    } //~ ERROR: expected
    function deposit() public payable {}
}
