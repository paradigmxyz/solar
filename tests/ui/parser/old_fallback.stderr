error: expected a state variable declaration
  --> ROOT/tests/ui/parser/old_fallback.sol:LL:CC
   |
LL |     function() external payable {
   |                                 ^
   |
   = note: this style of fallback function has been removed; use the `fallback` or `receive` keywords instead

error: expected one of `(`, `.`, `;`, `?`, `[`, `payable`, `pure`, `view`, or `{`, found `}`
  --> ROOT/tests/ui/parser/old_fallback.sol:LL:CC
   |
LL |         uint
   |             ^ expected one of 9 possible tokens
LL |     }
   |     ^ unexpected token
   |

error: aborting due to 2 previous errors

